src/test/java/ch/uzh/ifi/hase/soprafs23/controller/UserControllerTest.java

```diff
@@ -43,61 +48,136 @@ public class UserControllerTest {
-    userPostDTO.setName("Test User");
-    userPostDTO.setUsername("testUsername");
+    userPostDTO.setUsername(user.getUsername());
+    userPostDTO.setPassword(user.getPassword());
 
-    given(userService.createUser(Mockito.any())).willReturn(user);
+    // mocks the createUser(user) method in UserService
+    given(userService.createUser(Mockito.any(User.class))).willReturn(user);
 
-    // when/then -> do the request + validate the result
+    // when
     MockHttpServletRequestBuilder postRequest = post("/users")
-        .contentType(MediaType.APPLICATION_JSON)
-        .content(asJsonString(userPostDTO));
+                                                .contentType(MediaType.APPLICATION_JSON)
+                                                .content(asJsonString(userPostDTO));
 
     // then
     mockMvc.perform(postRequest)
         .andExpect(status().isCreated())
         .andExpect(jsonPath("$.id", is(user.getId().intValue())))
-        .andExpect(jsonPath("$.name", is(user.getName())))
         .andExpect(jsonPath("$.username", is(user.getUsername())))
         .andExpect(jsonPath("$.status", is(user.getStatus().toString())));
+
+    // verify that the correct calls on UserService were made
+    verify(userService, times(1)).createUser(Mockito.any(User.class));
   }
 
+
+  @Test
+  public void postUsers_returnsErrorConflict_whenInvalidInput() throws Exception {
+    // create UserPostDTO for the request body
+    UserPostDTO userPostDTO = new UserPostDTO();
+    userPostDTO.setUsername(user.getUsername());
+    userPostDTO.setPassword(user.getPassword());
+
+    // mocks the createUser(user) method in UserService
+    given(userService.createUser(Mockito.any(User.class)))
+        .willThrow(new ResponseStatusException(HttpStatus.CONFLICT, "error message"));
+
+    // when
+    MockHttpServletRequestBuilder postRequest = post("/users")
+                                                .contentType(MediaType.APPLICATION_JSON)
+                                                .content(asJsonString(userPostDTO));
+
+    // then
+    mockMvc.perform(postRequest)
+        .andExpect(status().isConflict());
+
+    // verify that the correct calls on UserService were made
+    verify(userService, times(1)).createUser(Mockito.any(User.class));
+  }
+
+    /**
+     * TEST PUT user by username - VALID version
+     */
+    @Test
+    public void loginUser_valid() throws Exception {
+        // create new User
+        User user = new User();
+        user.setId(3L);
+        user.setUsername("testUsername");
+
+        //don't need comparison with internal representation here
+
+        // when/then -> do the request - just trying to put (=update) sth.
+        MockHttpServletRequestBuilder putRequest = put("/users/testUsername/login")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(asJsonString(user));
+
+        // then
+        mockMvc.perform(putRequest)
+                .andExpect(status().isOk());
```

Comment:
the login mapping in UserController returns a userGetDTO of the user, maybe check for that content here as well, like it's checked in my postUsers_returnsCreatedUser_whenvalidInput test for example :)

Reply:
wrote new test loginUser_Test() and changed putUser_VALID